sklearn/exceptions.py

```diff
@@ -38,6 +39,19 @@ class NotFittedError(ValueError, AttributeError):
     """
 
 
+class NotSupportedByEngineError(NotImplementedError):
+    """External plugins might not support all the combinations of parameters and
+    input types that the the vanilla sklearn implementation otherwise supports. In such
```

Comment:
Thanks for adding the purpose and the specification of failing tests' semantic.
```suggestion
    input types that the vanilla sklearn implementation otherwise supports. In such
```

Reply:
note that @betatim also have a proposal for this, relying on an `accept` method exposed by engines rather than catching an exception.